### PR TITLE
feat(example): add drawing feature to the canvas example

### PR DESCRIPTION
**Pull request by orhun**
rec_20241018T235208

fun fact: I had to do [35 pushups](https://www.youtube.com/watch?v=eS92stzBYXA) for this...

### CODE REVIEW

File: examples/canvas.rs [hunk 1]
```diff
@@ -17,22 +17,24 @@ use std::time::{Duration, Instant};
...
-        canvas::{Canvas, Circle, Map, MapResolution, Rectangle},
+        canvas::{Canvas, Circle, Map, MapResolution, Points, Rectangle},
         Block, Widget,
     },
     DefaultTerminal, Frame,
 };
 
 fn main() -> Result<()> {
     color_eyre::install()?;
+    ratatui::crossterm::execute!(std::io::stderr(), crossterm::event::EnableMouseCapture)?;
```
Comment (joshka) [inline examples/canvas.rs:33]: style: add an import

File: examples/canvas.rs [hunk 2]
```diff
@@ -82,6 +89,19 @@ impl App {
                         KeyCode::Left | KeyCode::Char('h') => self.x -= 1.0,
                         _ => {}
                     }
+                } else if let Event::Mouse(event) = event {
```
Comment (joshka) [inline examples/canvas.rs:92]: style: use match for multiple values and extract the handlers to methods

**Review by joshka — APPROVED** [review]
Just style nits

**Comment by orhun** [comment]
Thanks for making the changes :)

**Pull request merged**